### ISSUE TITLE
"Current user does not have an active cart" even when he actually has one

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/magento2#26117]
### Preconditions (*)

1. Registered Customer

### Steps to reproduce (*)

1. Generate token
2. Create an empty cart
3. Get inactive cart through the order completion
4. Generate token again
5. Create an empty cart
6. Put a product to cart
7. Set Billing address with card_id of inactive cart

### Expected result (*)

1. Truth

### Actual result (*)

1. False
```graphql
{
  "errors": [
    {
      "message": "Could not find a cart with ID \"PeLzHtxg5Ujw98wyaRt8MCX6dcJggjwa\"",
      "category": "graphql-no-such-entity",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "setBillingAddressOnCart"
      ]
    }
  ],
  "data": {
    "setBillingAddressOnCart": null
  }
}
```
###  Additional confirmation info:
Use the GraphQL checkout tutorial if you need help with reproducing the issue 
https://devdocs.magento.com/guides/v2.3/graphql/tutorials/checkout/

**Comment by lenaorobei** [comment]
@TomashKhamlai please verify if this issue is reproducible.

**Comment by engcom-Bravo** [comment]
Reproducible on Magento 2.4
This error message appears:
```graphql
{
  "errors": [
    {
      "message": "Current user does not have an active cart."
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29936` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by slavvka** [comment]
Hi @TomashKhamlai. Thank you for your report. 
The issue has been fixed in magento/magento2#27187 by @engcom-Charlie in 2.4-develop branch
Related commit(s):
 - [afb752e629e96b35a1380d8c3909f7366eddfac6](https://github.com/magento/magento2/commit/afb752e629e96b35a1380d8c3909f7366eddfac6)
 - [1f87785fb1c7f70bc9cc1a947ed78dca7a781256](https://github.com/magento/magento2/commit/1f87785fb1c7f70bc9cc1a947ed78dca7a781256)
 - [a8450f4dd847c8e5af63b591579529943e555f29](https://github.com/magento/magento2/commit/a8450f4dd847c8e5af63b591579529943e555f29)

The fix will be available with the upcoming 2.4.0 release.
